### PR TITLE
Issue/13127 add edit sites button

### LINKED ISSUE #13127: Add `Edit` button to site picker when there are more than 1 site listed

### PR DESCRIPTION
**Pull request by JorgeMucientes**
Closes: #13127 

### Description

Adds new `Edit Stores` button to site picker when the list displays more than 1 site. Additionally, adds a feature flag to avoid showing this new button in production builds. 

### Testing information
1. Log into the app using WP.com credentials with an account that has multiple stores
2. Go to the site picker
3. Check the new button `Edit Stores` is displayed once the store list is loaded

### The tests that have been performed

The above

### Images/gif

https://github.com/user-attachments/assets/49b82575-68d0-49ef-abba-df3612e56baa

- [X] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/sitepicker/SitePickerViewModelTest.kt [hunk 1]
```diff
@@ -151,7 +151,7 @@ class SitePickerViewModelTest : BaseUnitTest() {
         userInfo = SitePickerTestUtils.userInfo,
         primaryBtnText = resourceProvider.getString(R.string.continue_button),
         secondaryBtnText = resourceProvider.getString(R.string.login_try_another_account),
-        hasConnectedStores = defaultExpectedSiteList.isNotEmpty()
+        hasConnectedStores = defaultExpectedSiteList.isNotEmpty(),
```
Comment (hafizrahman) [inline WooCommerce/src/test/kotlin/com/woocommerce/android/ui/sitepicker/SitePickerViewModelTest.kt]: very np: is this comma needed?

**Review by hafizrahman — APPROVED** [review]
This LGTM 👍🏼 👍🏼  

Just left a np comment.

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/245">21.3</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Pull request merged**